dask/dataframe/core.py

```diff
@@ -1546,6 +1546,31 @@ def sample(self, n=None, frac=None, replace=False, random_state=None):
             If int we create a new RandomState with this as the seed
             Otherwise we draw from the passed RandomState
 
+        !!! Limitation
+        --------------
+            Use only if frac * length(df) >= number_of_partitions
+            is satisfied.
+
+        Case 1:  frac * length(df) >= number_of_partitions
+            Sample works correctly and returns a sample of
+            sample_size = frac * length
+        Case 2: frac * length(df) < number_of_partitions
+            Sample does not work correctly and does not return
+            a sample of sample_size = frac * length
+
+        Example of limitation
+        ---------------------
+            import pandas as pd
+            import dask.dataframe as dd
+
+            df = dd.from_pandas(pd.DataFrame({'test':range(0,100)}),
+                                npartitions= 10)
+            print(f'Expectation | Result')
+            print(f'--------------------')
+            for numerator in (1, 5, 9, 10, 20, 30, 50):
+                sample = df.sample(frac= numerator/100)
+                print(f'{numerator:^11} | {len(sample):^7}')
```

Comment:
```suggestion
        Notes
        -----
        If `frac` is less than the number of partitions divided by the length
        of the object, you may see unexpected results. As an example:
        
        >>> import pandas as pd
        >>> import dask.dataframe as dd
        >>> df = pd.DataFrame({'test': range(0, 100)})
        >>> ddf = dd.from_pandas(df, npartitions=10)
        >>> numerators = [1, 9, 10, 20]
        >>> # expected length after sampling
        >>> numerators
        [1, 9, 10, 20]
        >>> # result of sample
        >>> [len(ddf.sample(frac=numerator/100)) for numerator in numerators]
        [0, 10, 10, 20]
```

Reply:
Looks good to me!

Reply:
On a side note, I think that a future resolution should involve shifting from an implicit (```frac```) to an explicit (```n```) sample size parameter. Within a framework like dask where attributes like length are only calculated on demand, explicit sample size is more natural.

Reply:
I completely agree that would be more intuitive!